src/main/java/friday/parser/Parser.java

```diff
@@ -0,0 +1,82 @@
+package friday.parser;
+
+import java.io.IOException;
+
+import friday.storage.Storage;
+import friday.task.TaskList;
+
+/**
+ * The Parser class is responsible for interpreting user input commands and executing corresponding actions.
+ * It interacts with the TaskList and Storage classes to manage tasks and data storage.
+ */
+public class Parser {
+    private TaskList tasks;
+
+    private Storage storage;
```

Comment:
Maybe you could also set some of these to be final as well if needed.

Reply:
alright! will change!